crates/router/src/connector/cashtocode/transformers.rs

```diff
@@ -0,0 +1,337 @@
+use common_utils::pii::Email;
+use masking::Secret;
+use router_env::logger;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils::{self},
+    core::errors,
+    services,
+    types::{self, api, storage::enums},
+};
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CashtocodePaymentsRequest {
+    amount: i64,
+    transaction_id: String,
+    user_id: String,
+    currency: String,
+    first_name: Option<String>,
+    last_name: Option<String>,
+    user_alias: String,
+    requested_url: String,
+    cancel_url: String,
+    email: Option<Secret<String, Email>>,
+    mid: String,
+}
+
+pub struct CashToCodeMandatoryParams {
+    pub user_id: String,
+    pub user_alias: String,
+    pub requested_url: String,
+    pub cancel_url: String,
+}
+
+fn get_mid(
+    payment_method_data: &api::payments::PaymentMethodData,
+) -> Result<String, errors::ConnectorError> {
+    match payment_method_data {
+        api_models::payments::PaymentMethodData::Reward(reward_data) => {
+            Ok(reward_data.mid.to_string())
+        }
+        _ => Err(errors::ConnectorError::NotImplemented(
+            "Payment methods".to_string(),
+        )),
+    }
+}
+
+fn get_mandatory_params(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<CashToCodeMandatoryParams, error_stack::Report<errors::ConnectorError>> {
+    let customer_id = item
+        .request
+        .customer_id
+        .as_ref()
+        .ok_or_else(utils::missing_field_err("customer_id"))?;
+    let url = item
+        .return_url
+        .to_owned()
+        .ok_or_else(utils::missing_field_err("return_url"))?;
+    Ok(CashToCodeMandatoryParams {
+        user_id: customer_id.to_owned(),
+        user_alias: customer_id.to_owned(),
+        requested_url: url.to_owned(),
+        cancel_url: url,
+    })
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for CashtocodePaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let params: CashToCodeMandatoryParams = get_mandatory_params(item)?;
+        let mid_helper = get_mid(&item.request.payment_method_data);
+        let mid = match mid_helper {
+            Ok(mid) => mid,
+            Err(err) => return Err(err.into()),
+        };
+        match item.payment_method {
+            storage_models::enums::PaymentMethod::Reward => Ok(Self {
+                amount: item.request.amount,
+                transaction_id: item.attempt_id.clone(),
+                currency: item.request.currency.to_string(),
+                user_id: params.user_id,
+                first_name: None,
+                last_name: None,
```

Comment:
what is the use of first_name and last_name if we are not using it

Reply:
We might use them in future